### ISSUE TITLE
GraphQL::Pro is not compatible with GraphQL Ruby 1.12.19

### ISSUE THREAD

**Issue by waissbluth** [rmosolgo/graphql-ruby#4696]
**Describe the bug**

When syncing persisted queries, there is a call made with two arguments to a method that takes only one.

**Versions**

`graphql` version: 1.12.19
`rails` (or other framework): 6.1.7.6
`graphql-pro` 1.24.10

**Steps to reproduce**

Call the `/graphql/sync` endpoint or `GraphQL::Pro::OperationStore::AddOperationBatch` directly

**Expected behavior**

No errors

**Actual behavior**

What specifically went wrong?

```
ArgumentError: wrong number of arguments (given 2, expected 1)
graphql-1.12.19/lib/graphql/schema/member/has_arguments.rb:97:in `get_argument'
```

<details>
<summary>Click to view exception backtrace</summary>

```
ArgumentError: wrong number of arguments (given 2, expected 1)
graphql-1.12.19/lib/graphql/schema/member/has_arguments.rb:97:in `get_argument'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:63:in `block in on_argument'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:106:in `block in each_field'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:105:in `each'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:105:in `each_field'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:59:in `on_argument'
graphql-1.12.19/lib/graphql/static_validation/rules/variable_usages_are_allowed.rb:49:in `on_argument'
graphql-1.12.19/lib/graphql/static_validation/rules/argument_literals_are_compatible.rb:8:in `on_argument'
graphql-1.12.19/lib/graphql/static_validation/rules/arguments_are_defined.rb:9:in `on_argument'
graphql-1.12.19/lib/graphql/static_validation/base_visitor.rb:137:in `on_argument'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `public_send'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `visit_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
graphql-1.12.19/lib/graphql/language/visitor.rb:102:in `block in on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `each'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:123:in `on_field'
/graphql-pro/lib/graphql/pro/operation_store/index_visitor.rb:49:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/argument_names_are_unique.rb:9:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/required_arguments_are_present.rb:7:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/fields_have_appropriate_selections.rb:12:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/fields_will_merge.rb:29:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/fields_are_defined_on_type.rb:25:in `on_field'
graphql-1.12.19/lib/graphql/static_validation/rules/unique_directives_per_location.rb:27:in `block (2 levels) in <module:UniqueDirectivesPerLocation>'
graphql-1.12.19/lib/graphql/static_validation/base_visitor.rb:106:in `on_field'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `public_send'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `visit_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
graphql-1.12.19/lib/graphql/language/visitor.rb:102:in `block in on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `each'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:123:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/definition_dependencies.rb:47:in `on_operation_definition'
/graphql-pro/lib/graphql/pro/operation_store/operation_names_are_present_and_unique.rb:21:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/subscription_root_exists.rb:12:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/mutation_root_exists.rb:12:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/variable_usages_are_allowed.rb:13:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/variables_are_used_and_defined.rb:44:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/variable_names_are_unique.rb:20:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/fields_have_appropriate_selections.rb:18:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/fields_will_merge.rb:24:in `on_operation_definition'
graphql-1.12.19/lib/graphql/static_validation/rules/unique_directives_per_location.rb:27:in `block (2 levels) in <module:UniqueDirectivesPerLocation>'
graphql-1.12.19/lib/graphql/static_validation/base_visitor.rb:76:in `on_operation_definition'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `public_send'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `visit_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
graphql-1.12.19/lib/graphql/language/visitor.rb:102:in `block in on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `each'
graphql-1.12.19/lib/graphql/language/visitor.rb:101:in `on_abstract_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:123:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/definition_dependencies.rb:38:in `on_document'
/graphql-pro/lib/graphql/pro/operation_store/operation_names_are_present_and_unique.rb:25:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/variables_are_used_and_defined.rb:79:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/fragment_spreads_are_possible.rb:26:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/fragments_are_used.rb:6:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/fragments_are_finite.rb:6:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/fragment_names_are_unique.rb:17:in `on_document'
graphql-1.12.19/lib/graphql/static_validation/rules/no_definitions_are_present.rb:34:in `on_document'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `public_send'
graphql-1.12.19/lib/graphql/language/visitor.rb:79:in `visit_node'
graphql-1.12.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
graphql-1.12.19/lib/graphql/language/visitor.rb:68:in `visit'
graphql-1.12.19/lib/graphql/static_validation/validator.rb:57:in `block (3 levels) in validate'
graphql-1.12.19/lib/graphql/static_validation/validator.rb:44:in `catch'
graphql-1.12.19/lib/graphql/static_validation/validator.rb:44:in `block (2 levels) in validate'
timeout-0.4.0/lib/timeout.rb:170:in `timeout'
graphql-1.12.19/lib/graphql/static_validation/validator.rb:43:in `block in validate'
graphql-1.12.19/lib/graphql/tracing.rb:81:in `call_tracers'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `block in call_tracers'
/Users/nwr/src/tines/backend/app/graphql/tracing/sentry_tracer.rb:14:in `trace'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `call_tracers'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `block in call_tracers'
/Users/nwr/src/tines/backend/app/graphql/tracing/rails_log_tracer.rb:14:in `trace'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `call_tracers'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `block in call_tracers'
/Users/nwr/src/tines/backend/app/graphql/tracing/honeycomb_graphql_tracer.rb:56:in `trace'
graphql-1.12.19/lib/graphql/tracing.rb:83:in `call_tracers'
graphql-1.12.19/lib/graphql/tracing.rb:67:in `trace'
graphql-1.12.19/lib/graphql/static_validation/validator.rb:30:in `validate'
/graphql-pro/lib/graphql/pro/operation_store/validate.rb:41:in `validate'
/graphql-pro/lib/graphql/pro/operation_store/active_record_backend.rb:48:in `block in batch_upsert_client_operations'
/graphql-pro/lib/graphql/pro/operation_store/active_record_backend.rb:47:in `map'
/graphql-pro/lib/graphql/pro/operation_store/active_record_backend.rb:47:in `batch_upsert_client_operations'
/graphql-pro/lib/graphql/pro/operation_store/add_operation_batch.rb:159:in `save!'
/graphql-pro/lib/graphql/pro/operation_store/add_operation_batch.rb:38:in `block in call'
/graphql-pro/lib/graphql/pro/operation_store/active_record_backend.rb:364:in `block in transaction'
activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
activerecord-6.1.7.6/lib/active_record/transactions.rb:209:in `transaction'
/graphql-pro/lib/graphql/pro/operation_store/active_record_backend.rb:362:in `transaction'
/graphql-pro/lib/graphql/pro/operation_store/add_operation_batch.rb:27:in `call'

```

</details>

**Additional context**

The following monkey-patch fixes the issue:

```
module GraphQL
  class Schema
    class Member
      module HasArguments
        old_get_argument = instance_method(:get_argument)

        # Version 1.12.19 of graphql-ruby takes one argument
        define_method(:get_argument) do |arg_defn, _context = nil|
          old_get_argument.bind_call(self, arg_defn)
        end
      end
    end
  end
end
```

**Comment by rmosolgo** [comment]
Thanks for the detailed report and for sharing the workaround you found. I just released graphql-pro v1.24.11 which includes a patch to send only _one_ argument when `get_argument` only takes one. Please give that new version a try and let me know if you have any more trouble with it!

**Comment by waissbluth** [comment]
Just upgraded and it works perfectly, thanks!
